src/network/networkd-dhcp-server.c

```diff
@@ -197,8 +197,14 @@ int address_acquire_from_dhcp_server_leases_file(Link *link, const Address *addr
                         lease_file,
                         &a,
                         &prefixlen);
-        if (r < 0)
+        if (r == -ENOENT)
                 return r;
+        if (r < 0)
+                return log_warning_errno(r, "Failed to load lease file %s: %s",
+                                         lease_file,
+                                         r == -ENXIO ? "expected JSON content not found" :
+                                         r == -EINVAL ? "invalid JSON" :
+                                         STRERROR(r));
```

Comment:
Why not to use `%m`??

Reply:
Because the errors that are given by strerror (%m) are not comprehensible. E.g. ENXIO is "No such device or address", which would be particularly confusing in this context.

Reply:
Ah, STRERROR() is used conditionally. OK, LGTM.